### PR TITLE
Add ETag support

### PR DESCRIPTION
**Pull request by julesrenaud**
Hi @patrixr,

So I added basic support for conditional GETs using ETags. I've done some tests and it works as expected on my end.

As proposed, `enableEtagSupport: true` is used in the middleware settings to enable the feature.

I tried to follow the coding style as much as I could, let me know what you think!

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -148,6 +150,19 @@ const Cache = (strapi) => {
 
           const cacheEntry = await cache.get(cacheKey);
 
+          if (enableEtagSupport) {
+            const ifNoneMatch = ctx.request.headers['if-none-match'];
+            const etagEntry = await cache.get(`${cacheKey}_etag`);
+            const etagMatch = ifNoneMatch === etagEntry;
+
+            if (!etagMatch) {
+              ctx.response.etag = etagEntry;
```
Comment (patrixr) [inline lib/index.js:159]: I think `ctx.set('Etag', etagEntry)` is the recommended way to set headers

**Review by patrixr — APPROVED** [review]
Thanks @julesrenaud ! This looks good, just added one minor comment !

Couple of things:
- Could you bump the version's minor in package.json, that will allow github actions to auto-release to npm
- If possible, could you add a test case for this

**Comment by julesrenaud** [comment]
@patrixr I'll add the tests today, just need to familiarize myself with the test procedure

**Comment by patrixr** [comment]
Sure thing, the test setup was very quickly improvised, it's far from ideal and I do hope to rewrite them better one day

**Comment by julesrenaud** [comment]
@patrixr I added a couple tests to the cache.test.js file. Does it cover enough of the new feature in your opinion?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "strapi-middleware-cache",
-  "version": "1.2.0",
+  "version": "1.2.1",
```
Comment (patrixr) [inline package.json:3]: Since this is a new feature we can probably bump to `1.3.0`

**Comment by patrixr** [comment]
@julesrenaud I think the tests overall could do with some improvements (not yours specifically), the whole thing was very improvised since the module started as a small unambitious side project.

I do plan on re-visiting them sooner than later. For the meantime I think your tests should at least smoke out some regressions. If you could fix the conflict then I'll merge the PR

(btw, I think we can bump version to `1.3.0`, it's a significant improvement)

And thanks again for this !

**Comment by julesrenaud** [comment]
> (btw, I think we can bump version to `1.3.0`, it's a significant improvement)

Of course, I erred on this one

**Comment by julesrenaud** [comment]
@patrixr I think we should be good to go. I swapped double quotes for single quotes while merging cache.test.js (from the PR for caching headers)

**Comment by julesrenaud** [comment]
@patrixr are you going to take care of updating the README.md file?

**Comment by patrixr** [comment]
Great, yes I'll merge/tag and update the readme today ! Thank you !

**Review by patrixr — APPROVED** [review]

**Comment by patrixr** [comment]
@julesrenaud Version 1.3.0 is now available on NPM, cheers ! 👍

**Comment by julesrenaud** [comment]
> @julesrenaud Version 1.3.0 is now available on NPM, cheers ! 👍 

@patrixr glad I could help, thanks for the guidance!

**Pull request merged**